frontend/src/components/MilestoneCompletedness.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { Trans } from 'react-i18next';
+import Tooltip from '@mui/material/Tooltip';
+import { Task } from '../redux/roadmaps/types';
+import { BarSection } from './PercentageBar';
+import { TaskStatus } from '../../../shared/types/customTypes';
+import { taskStatusToText } from '../utils/TaskUtils';
+import { percent } from '../utils/string';
+import colors from '../colors.module.scss';
+import css from './MilestoneCompletedness.module.scss';
+
+const classes = classNames.bind(css);
+
+const colorByStatus = (status: number) => {
+  if (status === TaskStatus.COMPLETED) return colors.emerald;
+  if (status === TaskStatus.IN_PROGRESS) return colors.orange;
+  return colors.black10;
+};
+
+export const MilestoneCompletedness: FC<{
+  tasks: Task[];
+}> = ({ tasks }) => {
+  const total = tasks.length;
```

Comment:
```suggestion
  const total = tasks.length || 0;
```

This could also prevent some divisions by zero. And when there are no tasks dividing 0 / 1 gives quite reasonable 0.